### ISSUE TITLE
Invalid ingress class: IngressClass.networking.k8s.io "alb" not found

### ISSUE THREAD

**Issue by owyah** [kubernetes-sigs/aws-load-balancer-controller#2813]
**Describe the bug**
I got this error when apply https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/v2.4.3/docs/examples/2048/2048_full.yaml

**Steps to reproduce**
I follow this tutorial:
- https://www.youtube.com/watch?v=TUS8KWaGKco
- https://aws.amazon.com/premiumsupport/knowledge-center/eks-alb-ingress-controller-setup/

**Expected outcome**
successful install **ingress**

**Environment**
* AWS Load Balancer controller version: **v2.4.3**
* Kubernetes version: **Major:"1", Minor:"22+", GitVersion:"v1.22.11-eks-18ef993"**
* Using EKS version: **v1.22.12-eks-ba74326**

**Additional Context**:
Is there any setup step that I missed?

**Comment by kishorj** [comment]
@owyah, did you install the controller via manifests instead of helm chart? If so, can you verify you have the ingress class named alb in your cluster?

You can apply the following manifest for ingress class and params:
https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/download/v2.4.3/v2_4_3_ingclass.yaml

**Comment by owyah** [comment]
@kishorj I use manifests, based on the tutorial. I do not know yet about helm chart. I confused because to many manifest that should I apply. 
So, I just apply that manifest, right? let me try as soon as possible.
I just curious, maybe it is not good question, why this manifest:
https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/download/v2.4.3/v2_4_3_full.yaml
did not include:
https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/download/v2.4.3/v2_4_3_ingclass.yaml
Since it is full version of it. It confused me as a newbie.
**Thank you for your response**.

**Comment by kishorj** [comment]
In v2.4.3 patch release, we separated the IngressClass and IngressClassParams from the full manifest due to race condition with the kubectl apply - issue #2672.

Starting recent v2.4.4 patch release, we stripped down the IngressClassParams reference from the IngressClass and included in the full manifest. You should be apply just the full manifest in this case - https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/download/v2.4.4/v2_4_4_full.yml

**Comment by kishorj** [comment]
@owyah, I'm closing the issue since recent releases have a separate manifest for IngressClassParams.

**Comment by grahamschuckman** [comment]
Also had this same issue stemming from the AWS tutorial and documentation. Using the latest available version (currently 2.6.2) causes a failure with creating the alb ingress class. You have to use 2.4.1 in the video for this to work. Please update the documentation in the video tutorial and blog post here accordingly:

https://repost.aws/knowledge-center/eks-alb-ingress-controller-setup#

**Comment by BenAur** [comment]
Hello, adding a comment for people like me who find themselves on this issue while not understanding why setting a cluster with the file in version 2.4.1 was working but not in subsequent versions (like the version 2.4.7 for me)

The `alb` ingress class params and ingress class are not in the full manifest anymore. 
I fixed it by applying them manually as follow:

```sh
kubectl apply -f - <<EOF
apiVersion: elbv2.k8s.aws/v1beta1
kind: IngressClassParams
metadata:
  labels:
    app.kubernetes.io/name: aws-load-balancer-controller
  name: alb
---
apiVersion: networking.k8s.io/v1
kind: IngressClass
metadata:
  labels:
    app.kubernetes.io/name: aws-load-balancer-controller
  name: alb
spec:
  controller: ingress.k8s.aws/alb
  parameters:
    apiGroup: elbv2.k8s.aws
    kind: IngressClassParams
    name: alb
EOF
```

This fixed it for me.
